x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -246,7 +311,7 @@ export function getWaterfall(
     };
   }
 
-  const waterfallItems = trace.items.map(traceItem => {
+  const waterfallItems: IWaterfallTraceItem[] = trace.items.map(traceItem => {
```

Comment:
Then I force `waterfallItems` to be the Trace type, because I know that it will never be an AgentMark.